src/App.java

```diff
@@ -1,14 +1,143 @@
 
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.text.Format;
+import java.text.SimpleDateFormat;
+import java.time.Duration;
+import java.util.Date;
 import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
 import java.util.stream.Collectors;
 
 import org.openqa.selenium.By;
 import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.WebElement;
 import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.chrome.ChromeOptions;
 import org.openqa.selenium.support.ui.Select;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.JavascriptExecutor;
 
 public class App {
+
+    public static String[] pullVehicleSpecs(WebDriver vehicleDriver) {
+
+        String[] pulledSpecs = new String[12];
+
+        WebDriverWait w = new WebDriverWait(vehicleDriver, Duration.ofSeconds(3));
+        w.until(ExpectedConditions.presenceOfElementLocated(By.xpath("//ul[@id='sl-card-body']")));
+
+        JavascriptExecutor js = (JavascriptExecutor) vehicleDriver;
+
+        js.executeScript("document.getElementById('sl-card-body').classList.remove('closed')");
+
+        WebDriverWait w2 = new WebDriverWait(vehicleDriver, Duration.ofSeconds(3));
+        w2.until(ExpectedConditions.presenceOfElementLocated(By.id("specificationWidget")));
+
+        List<WebElement> specSpans = vehicleDriver.findElement(By.id("specificationWidget"))
+                .findElements(By.xpath("//span[starts-with(@id,'spec-key-')]"));
+
+        AtomicInteger specCounter = new AtomicInteger(0);
+
+        specSpans.forEach(spec -> {
+
+            if (spec.getText().equals("Kilometres")) {
```

Comment:
not a biggie, but I guess to use switch would fit this task more, since  spec.getText() would match one of the texts so it doesn't need to check the text value one by one